mozilla/ichnaea
# redo configuration with Everett

This redoes configuration across the application:

* changes some module names because "configuration" is ambiguous
* changes how settings are used so as to easily differentiate between settings and Python module "constants"
* moves non-settings out of the settings file
* collapse the DB settings into three non-composed DB URI settings
* remove some uses of the TESTING variable--we have a `test.env` file for testing-related settings
* update configuration documentation
* remove some monkey patching in the tests in favor of Everett's `config_override`

To test:

1. run `make build` then `make setup` to suss out issues in building and setting up services
2. run `make run` and go to `http://location:8000` and click around
3. run `make celery` and wait a while to see if it kicks up any errors

Fixes #836
